admin_console/controllers/UserController.php

```diff
@@ -193,14 +209,68 @@ public function loginAction()
 		$this->view->resetForm = $resetForm;
 		$this->render('login');
 	}
+
+	protected function ssoLogin($settings)
+	{
+		$partnerId = $settings->partnerId;
+		$client = Infra_ClientHelper::getClient();
+		$ssoPlugin = Kaltura_Client_Sso_Plugin::get($client);
+
+		try
+		{
+			$redirectUrl = $ssoPlugin->sso->login('', 'admin_console', $partnerId);
+			if(!$redirectUrl)
+			{
+				throw new Exception('Missing authentication server url', self::MISSING_AUTH_SERVER_URL);
+			}
+
+			// if we got session from sso server validate it, if we didnt redirect to sso server
+			$ks = $_GET['ks'];
+			if(!$ks)
+			{
+				$this->getResponse()->setRedirect($redirectUrl);
+			}
+			else
+			{
+				$client->setKs($ks);
+				$client->user->loginByKs($partnerId);
+
+				$adapter = new Kaltura_AdminAuthAdapter();
+				$adapter->setKs($ks);
+				$auth = Infra_AuthHelper::getAuthInstance();
+				$result = $auth->authenticate($adapter);
+				if ($result->isValid())
+				{
+					$this->_helper->redirector('list', 'partner');
+				}
+				else
+				{
+					throw new Exception('Could not authenticate with the given credentials', self::INVALID_CREDENTIALS);
+				}
+			}
+		}
+		catch(Exception $ex)
+		{
+			throw $ex;
+		}
+	}
 	
 	public function logoutAction()
 	{
 		Zend_Session::forgetMe();
 		$client = Infra_ClientHelper::getClient();
 		$client->session->end();
 		Infra_AuthHelper::getAuthInstance()->clearIdentity();
-		$this->_helper->redirector('index', 'index');
+
+		$settings = Zend_Registry::get('config')->settings;
```

Comment:
Since this code repeats twice (and may be more in the future), create a small function - shouldUseSso()